### PR TITLE
images: Build centos-8-stream from cloud image

### PR DESCRIPTION
**Pull request by martinpitt**
This avoids having to download the > 9 GB DVD, the cloud image is only
700 MB. The DVD also configures the wrong repositories, it uses standard
CentOS 8 instead of -stream. This avoids the version mixes that caused
the GnuTLS uninstallability and the sssd/kvdo version mismatches.

In other words, this gives us an *actual* Stream image now, with newer versions.

 * [x] image-refresh centos-8-stream
 * [x] Fix testExpired
 * [x] Fix check-metrics swap failures: https://github.com/cockpit-project/cockpit/pull/14652
 * [x] Fix `check-metrics TestMetrics.testPcpHistory`
 * [x] Fix check-storage-luks: https://bugzilla.redhat.com/show_bug.cgi?id=1881829 - fixed by https://github.com/cockpit-project/cockpit/pull/14658
 * [x] Backport the above fixes to rhel-8.3 branch: https://github.com/cockpit-project/cockpit/pull/14660
 * [x] Fix LUKS harder: https://github.com/cockpit-project/cockpit/pull/14662
 * [x] Backport ^ to rhel-8.3: https://github.com/cockpit-project/cockpit/pull/14663
 * [x] Fix boot hang on NetworkManager-wait-online.service
 * [x] Fix `test/verify/check-multi-machine TestMultiMachine.testFIPS`: https://github.com/cockpit-project/cockpit/pull/14666

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh in progress on 2-cockpit-9.
Log: https://logs.cockpit-project.org/logs/image-refresh-1250-20200922-090505/

**Comment by cockpituous** [comment]
image-refresh centos-8-stream done: https://github.com/cockpituous/bots/commits/image-refresh-centos-8-stream-20200922-093333

**Comment by martinpitt** [comment]
Ugh, missing libssh in the mock env..

**Comment by martinpitt** [comment]
The missing libssh-devel is fallout from https://github.com/cockpit-project/cockpit/commit/cc34ffcab0cee ; I pushed a fix.

**Comment by cockpituous** [comment]
image-refresh in progress on 4-ci-srv-03.
Log: https://logs.cockpit-project.org/logs/image-refresh-1250-20200922-100606/

**Comment by cockpituous** [comment]
image-refresh centos-8-stream done: https://github.com/cockpituous/bots/commits/image-refresh-centos-8-stream-20200922-103434

**Comment by martinpitt** [comment]
Still [lots of trouble](https://logs.cockpit-project.org/logs/pull-1250-20200922-104511-03645d4e-centos-8-stream-cockpit-project-cockpit/log.html), another round.

**Comment by cockpituous** [comment]
image-refresh in progress on 3-ci-srv-05.
Log: https://logs.cockpit-project.org/logs/image-refresh-1250-20200922-120404/

**Comment by cockpituous** [comment]
image-refresh centos-8-stream done: https://github.com/cockpituous/bots/commits/image-refresh-centos-8-stream-20200922-123333

**Comment by martinpitt** [comment]
This is quite ridiculous.. PAM behaves slightly different here, asking again for the current password when encountering an expired one. That by itself is fairly easy to fix:
```diff
--- test/verify/check-login
+++ test/verify/check-login
@@ -296,6 +296,15 @@ account    required     pam_succeed_if.so user ingroup %s""" % m.get_admin_group
         b.wait_visible("#conversation-group")
         b.wait_not_visible("#password-group")
         b.wait_not_visible("#user-group")
+        # some PAM configs require typing the old password again
+        try:
+            with b.wait_timeout(5):
+                b.wait_in_text("#conversation-prompt", "Current password")
+            b.set_val('#conversation-input', 'foobar')
+            b.click('#login-button')
+        except Error:
+            pass
+
         b.wait_in_text("#conversation-prompt", "New password")
         b.set_val('#conversation-input', '123foobar!@#')
         b.click('#login-button')
```

But then it fails later on with an "internal error: passwords do not match" instead of showing that in the conversation prompt. 

The [metrics failures](https://logs.cockpit-project.org/logs/pull-1250-20200922-124038-f437a084-centos-8-stream-cockpit-project-cockpit/log.html#125-2) are due to the cloud image having no swap, so we need to adjust the tests for that.

The tang failures are because of [cryptsetup triggering the OOM killer](https://logs.cockpit-project.org/logs/pull-1250-20200922-124038-f437a084-centos-8-stream-cockpit-project-cockpit/TestStorageLuks-testClevisTang-centos-8-stream-127.0.0.2-2501-FAIL.log.gz), that seems like a genuine bug in C8S.

The [rhel-8.3 branch](https://logs.cockpit-project.org/logs/pull-1250-20200922-124038-f437a084-centos-8-stream-cockpit-project-cockpit-rhel-8.3/log.html) fails even more hilariously, mostly the check-realms bits. That may just be the "reinit global machines" bug that @mvollmer is fixing ATM.

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-11.
Log: https://logs.cockpit-project.org/logs/image-refresh-1250-20200923-075656/

**Comment by cockpituous** [comment]
image-refresh centos-8-stream done: https://github.com/cockpituous/bots/commits/image-refresh-centos-8-stream-20200923-082424

**Comment by martinpitt** [comment]
Found the reason for the failing testPcpHistory -- the VM's own host name was not resolvable. One more round, hopefully the last one!

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-1250-20200923-143030/

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-8.
Log: https://logs.cockpit-project.org/logs/image-refresh-1250-20200923-144343/

**Comment by cockpituous** [comment]
image-refresh centos-8-stream done: https://github.com/cockpituous/bots/commits/image-refresh-centos-8-stream-20200923-161616

**Comment by martinpitt** [comment]
Some more failures.. [testRaid](https://logs.cockpit-project.org/logs/pull-1250-20200923-194827-5e74ce4a-centos-8-stream-cockpit-project-cockpit/log.html#250-2) just looks like a flake.

[testUsed](https://logs.cockpit-project.org/logs/pull-1250-20200923-194827-5e74ce4a-centos-8-stream-cockpit-project-cockpit/log.html#273-2) is another instance of cryptsetup OOMing, but as that's a `@nondestructive` test, we could just reduce its memory demands. That's a cryptsetup call that the test controls, we don't rely on the builtin cockpit one. I sent PR #14662 for this and another place.

**Comment by martinpitt** [comment]
Now there is a [testFIPS failure](https://logs.cockpit-project.org/logs/pull-1250-20200924-115552-5e74ce4a-centos-8-stream-cockpit-project-cockpit/log.html#131-2) which didn't happen before. But it [flaked twice](https://logs.cockpit-project.org/logs/pull-1250-20200924-115233-5e74ce4a-centos-8-stream-cockpit-project-cockpit-rhel-8.3/log.html#47-1) on rhel-8.3 as well, so let's block on that. I can reproduce.

**Comment by martinpitt** [comment]
OMG, all the green! :four_leaf_clover:

File: images/scripts/centos-8-stream.bootstrap [hunk 1]
```diff
@@ -1,11 +1,7 @@
...
+URL=https://cloud.centos.org/centos/8-stream/x86_64/images/
+LATEST_DAILY=$(curl -s "$URL" | sed -n '/<a href=.*GenericCloud.*qcow2/ { s/^.*href="//; s_".*$__; p }')
 
-# virt-install: using the url "http://mirror.centos.org/centos/8-stream/BaseOS/x86_64/os/" as --location is broken
-
-CUR=$(curl --show-error --silent https://mirrors.mit.edu/centos/8-stream/isos/x86_64/ | grep -o '"CentOS-Stream.*-dvd1.iso"' | sed 's/"//g')
-wget --continue https://mirrors.mit.edu/centos/8-stream/isos/x86_64/$CUR -O "$BASE/centos-8-stream.iso"
-$BASE/virt-install-fedora "$1" x86_64 "$BASE/centos-8-stream.iso"
-rm "$BASE/centos-8-stream.iso"
+exec $(dirname $0)/lib/debian.bootstrap "$1" "$URL/$LATEST_DAILY"
```
Comment (mvollmer) [inline images/scripts/centos-8-stream.bootstrap:7]: It's weird to see Debian mentioned here, but there is nothing Debian specific about lib/debian.bootstrap, so what about renaming it to lib/cloud.bootstrap?

**Review by mvollmer — APPROVED** [review]

File: images/scripts/centos-8-stream.bootstrap (hunk 1 above)
Comment (martinpitt) [reply images/scripts/centos-8-stream.bootstrap:7]: Ah, haha -- I actually wanted to do this, but it got lost in transl^Wrebasing. I'll send a follow-up for this!

Comment (martinpitt) [reply images/scripts/centos-8-stream.bootstrap:7]: PR #1260

**Pull request merged**